quickwit-cluster/src/cluster.rs

```diff
@@ -206,6 +219,22 @@ impl Cluster {
                 .node_state(&NodeId::from(member.clone()))
                 .unwrap();
 
+            let available_services = node_state
+                .get(AVAILABLE_SERVICES_KEY)
+                .ok_or_else(|| {
+                    anyhow::anyhow!(
+                        "Could not find `{}` key on node `{}` state.",
+                        AVAILABLE_SERVICES_KEY,
+                        member.internal_id()
+                    )
+                })
+                .map(|services_str| {
+                    services_str
+                        .split(',')
+                        .map(QuickwitService::try_from)
+                        .collect::<Result<Vec<_>, _>>()
+                })??;
```

Comment:
The logic here is needlessly complicated.
Returning an error if the key is absent is an error... but assuming this is what you wanted, this is a very convoluted way to code this.

```
 let available_services_val = node_state
        .get(AVAILABLE_SERVICES_KEY)
        .ok_or_else(|| ...)?;
let services = parse_available_services_val(available_services_val)?;
```

is nicer for the reader.